docs/build-the-builder.md

```diff
@@ -0,0 +1,186 @@
+# Build The Builder Container
+
+The KubeVirt build system starts with a builder helper container, where the entire toolchain and its dependencies are stored, in order to
+ensure consistency between builds. Yet, there are sometimes CVEs that must be remedied with an update of the build toolchain, and from time
+to time, KubeVirt must be enabled for a new system architecture, e.g. `ppc64le` or `s390x`. 
+
+The KubeVirt official build stores its builder container and built images in a protected quay.io registry. If there is no builder image for the architecture
+of the machine you are trying to build KubeVirt on and for, `make cluster-up` will fail in its attempt to pull a builder image. You will need to build your own builder image, and tweak the build scripts to refer to it. 
+
+
+This document will provide the steps to:
+ - prepare the local environment to build the builder container
+ - how to update the rpm dependencies with bazeldnf
+ - build the builder container
+ - push the builder image to a custom registry
+ - invoke the new test builder image to build new kubevirt operator images
+ - push the new kubevirt operator images to a custom registry
+
+
+## Prerequisites
+
+### Podman or Docker
+
+install podman or docker
+
+On Ubuntu,
+
+```
+$ apt get install -y podman
+```
+
+or 
+
+```
+$ apt get install -y docker
+```
+
+On RHEL or CENTOS
+
+```
+$ dnf install -y podman
+```
+
+### Configure your read/write access to a container registry
+
+You will first need to create read and write access to an appropriate namespace in a container registry
+
+[`docker.io`](https://docs.docker.com/docker-id/) or [`quay.io`](https://docs.quay.io/solution/getting-started.html) or [`icr.io`](https://cloud.ibm.com/docs/Registry?topic=Registry-getting-started) are popular registries to use for this purpose.  Then set up a repository or namespace to hold the container you are building, e.g. `builder` or, where the namespace needs to be more unique, `kubevirt-builder`. 
+
+##  Install bazel and dependencies in order to containerize the builder 
+
+Bazel binary distribution images for version 5.3.0 for the Arm and x86 architectures are available from https://github.com/bazelbuild/bazel/releases/tag/5.3.0
```

Comment:
I'd add a note that you can also use [bazelisk](https://github.com/bazelbuild/bazelisk#bazelisk), which automatically downloads and installs the bazel release you need.

See the [.bazelversion](https://github.com/kubevirt/kubevirt/tree/main/.bazelversion) file.

Reply:
great.  It will be nice when essential tools such as bazel, bazelisk and bazeldnf have regular binary releases for all the architectures we need to use them for.

Reply:
Understood that this will not work in arm case. Sorry for my ignorance.

Reply:
so, when bazelisk can automatically download and install the bazel releases for s390x architecture, we can modify this build-the-builder.md

Reply:
trying to figure out how to 'resolve conversation' here